modules/src/clients/host_functions.rs

```diff
@@ -62,7 +63,14 @@ pub trait HostFunctionsProvider: Clone + Send + Sync + Default {
 /// This is a work around that allows us to have one super trait [`HostFunctionsProvider`]
 /// that encapsulates all the needed host functions by different subsytems, and then
 /// implement the needed traits through this wrapper.
-#[derive(Clone, Debug, Default)]
+#[derive(Derivative)]
+#[derivative(
+    Debug(bound = ""),
```

Comment:
what does `bound = ""` mean?

Reply:
It means that no additional bounds are needed when implementing the trait (i.e. no `T: Debug`)

Reply:
It gives the ability to explicitly specify the bounds. If one were to use `Derive` from std, there would be no such option, and the `T: Debug` bound would still be placed - we don't need that.